presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSimplifyExpressions.java

```diff
@@ -86,6 +86,8 @@ public void testExtractsCommonPredicate()
         assertSimplifies("((X OR V) AND Z) OR ((X OR V) AND V)", "(((X OR V) AND Z) OR V)");
         assertSimplifies("X AND ((Y AND Z) OR (Y AND V) OR (Y AND X))", "X AND Y AND (Z OR V OR X)");
         assertSimplifies("(A AND B AND C AND D) OR (A AND B AND E) OR (A AND F)", "A AND ((B AND C AND D) OR (B AND E) OR F)");
+
+        assertSimplifies("((A or B) and (A or C)) or D", "A or (B and C) or D");
```

Comment:
add similar test for reversed types of logical operators:
```
assertSimplifies("(((A AND B) OR (A AND C)) AND D) OR E", "(A AND (B OR C) AND D) OR E");
```
maybe also test for deeper more nested expression:
```
assertSimplifies("(((A or B) and (A or C)) or D) AND E", "(A or (B and C) or D) AND E");
```